### PR TITLE
Added tests for do_path_action

### PR DESCRIPTION
**Pull request by SnowballAntrobus**
We made multiple tests for do_path_action, covering all possible outcomes of the function in order to insure that it works correctly. Since the functions at this time are still fairly simple, the outputs of this tests will show whether they are printing the correct strings or not. This is the PR for issue #181 .

### CODE REVIEW

**Comment by SnowballAntrobus** [comment]
Of course, we now need to check if it compiles post GS PR #186

File: src/action_management/tests/test_path_action.c [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+#include "actionmanagement.h"
+#include "action_structs.h"
+#include "game.h"
+#include "path.h"
+
+bool execute_do_path_action(char *c_name, enum action_kind kind)
+{
+    char *dest = room_new("dummyroom", "a dummy room", "a placeholder room");
+    char *direction = "south";
+    game_t *g = game_new;
```
Comment (MTUC08) [inline src/action_management/tests/test_path_action.c]: Should this be game_new()?

File: src/action_management/tests/test_path_action.c [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    
+
+    char *expected_output = malloc(100); // buffer
+    expected_output[0] = '\0';
+    strcat(expected_output, "Requested action ");
+    strcat(expected_output, c_name);
+    strcat(expected_output, " in direction ");
+    strcat(expected_output, p->direction);
+    strcat(expected_output, " into room");
+    strcat(expected_output, p->dest);
```
Comment (MTUC08) [inline src/action_management/tests/test_path_action.c]: This is a carry-over typo from before I changed it in actions/dev, it should be p->dest->room_id.

**Comment by MTUC08** [comment]
Note: make sure you don't merge to actions/dev until #244 goes through

File: src/action_management/tests/test_path_action.c (hunk 1 above)
Comment (SnowballAntrobus) [reply src/action_management/tests/test_path_action.c]: haha well it doesn't matter now but actually that is valid in C (CLI taught me this)

**Comment by SnowballAntrobus** [comment]
Everything compiles and all tests pass

**Comment by SnowballAntrobus** [comment]
Closing statement:
We wrote basic tests for the function do_path_action. These tests focused on making sure that the correct string is being printed when correct and incorrect action kinds are used. In future work we would make testing simpler by making the return value of the function an error code int. Additionally, we will make more tests for this function once it is updated to change game state.

**Comment by mpcsh** [comment]
Issue Score: ✓+

Comments:
Good work, but the closing statement only came well after the fact.

**Pull request merged**